src/rustinfo.rs

```diff
@@ -115,6 +115,35 @@ fn load_setup(rust_info: &mut RustInfo) {
     };
 }
 
+fn load_triple(rust_info: &mut RustInfo) {
+    let mut triple = String::new();
+
+    if let Some(arch) = &rust_info.target_arch {
+        triple.push_str(arch);
+    } else {
+        triple.push_str("unknown")
+    }
+
+    if let Some(vendor) = &rust_info.target_vendor {
+        triple.push_str("-");
+        triple.push_str(vendor);
+    }
+
+    if let Some(os) = &rust_info.target_os {
```

Comment:
should we set triple if this is none but env is with info?

Reply:
I don't know if there is a standard to this, I tried looking for it.
But if we look at the [official rust platform support list](https://forge.rust-lang.org/release/platform-support.html), apparently the only mandatory things are `arch` and `os`.
But taking a look in the [llvm source](https://llvm.org/doxygen/Triple_8h_source.html), `arch`, `vendor` and `os` are mandatory.

I will go ahead and follow rust here.

Reply:
Should be better now.